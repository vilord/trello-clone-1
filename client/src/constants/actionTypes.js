/**
 * USER
 */
export const SIGNUP_USER_REQUEST = 'SIGNUP_USER_REQUEST';
export const SIGNUP_USER_SUCCESS = 'SIGNUP_USER_SUCCESS';
export const SIGNUP_USER_FAILURE = 'SIGNUP_USER_FAILURE';

export const LOGIN_USER_REQUEST = 'LOGIN_USER_REQUEST';
export const LOGIN_USER_SUCCESS = 'LOGIN_USER_SUCCESS';
export const LOGIN_USER_FAILURE = 'LOGIN_USER_FAILURE';

export const LOGOUT_USER = 'LOGOUT_USER';

export const GET_USER_SESSION_REQUEST = 'GET_USER_SESSION_REQUEST';
export const GET_USER_SESSION_SUCCESS = 'GET_USER_SESSION_SUCCESS';

export const SET_USER_PROFILE_REQUEST = 'SET_USER_PROFILE_REQUEST';
export const SET_USER_PROFILE_SUCCESS = 'SET_USER_PROFILE_SUCCESS';
export const SET_USER_PROFILE_FAILURE = 'SET_USER_PROFILE_FAILURE';

/**
 * UI
 */
export const SET_UI_ERROR = 'SET_UI_ERROR';
export const RESET_UI_ERROR = 'RESET_UI_ERROR';

export const SHOW_MAIN_MENU = 'SHOW_MAIN_MENU';
export const HIDE_MAIN_MENU = 'HIDE_MAIN_MENU';

export const SHOW_BOARDS_MENU = 'SHOW_BOARDS_MENU';
export const HIDE_BOARDS_MENU = 'HIDE_BOARDS_MENU';

export const SHOW_SEARCH_MENU = 'SHOW_SEARCH_MENU';
export const HIDE_SEARCH_MENU = 'HIDE_SEARCH_MENU';

export const SHOW_CREATE_MENU = 'SHOW_CREATE_MENU';
export const HIDE_CREATE_MENU = 'HIDE_CREATE_MENU';

export const SHOW_USER_MENU = 'SHOW_USER_MENU';
export const HIDE_USER_MENU = 'HIDE_USER_MENU';

export const FOCUS_HEADER_SEARCH = 'FOCUS_HEADER_SEARCH';
export const BLUR_HEADER_SEARCH = 'BLUR_HEADER_SEARCH';

/**
 * Cards
 */
export const SHOW_CARD_DETAIL = 'SHOW_CARD_DETAIL ';
export const HIDE_CARD_DETAIL = 'HIDE_CARD_DETAIL ';

export const SHOW_LABEL_TEXT = 'SHOW_LABEL_TEXT';
export const HIDE_LABEL_TEXT = 'HIDE_LABEL_TEXT';
