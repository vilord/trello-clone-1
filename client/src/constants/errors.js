export const invalidEmail = {
  kind: 'invalidEmail',
  header: 'Verify email!',
  message: "That doesn't look like an email addressâ€¦",
};

export const shortPassword = {
  kind: 'shortPassword',
  header: 'Password is too short!',
  message: 'Your password must be at least 8 characters long.',
};

export const userExists = {
  kind: 'userExists',
  header: 'User already exists!',
  message: 'Email is already in use by another account.'
};

export const wrongCredentials ={
  kind: 'wrongCredentials',
  header: 'Something wrong!',
  message: 'Incorrect email (username) or password.'
};

export const serverError = {
  kind: 'serverError',
  header: 'Unknown Server Error!',
  message: 'Our server may be down. Please retry later!'
};
