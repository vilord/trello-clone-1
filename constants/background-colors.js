module.exports = {
  BLUE: '#127ABD',
  ORANGE: '#D18F3E',
  GREEN: '#53973F',
  RED: '#AE4736',
  PURPLE: '#88629C',
  PINK: '#CB5C91',
  LIGHT_GREEN: '#50BE6E',
  LIGHT_BLUE: '#1CAECA',
  GREY: '#838C91',
};
