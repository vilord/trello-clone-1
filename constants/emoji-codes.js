module.exports = [
  'thumbsup',
  'thumbsdown',
  'raised_hands',
  'pray',
  'eyes',
  'rocket',
  'joy',
  'blush',
  'hearth_eyes',
  'throphy',
  'heart',
  'broken_heart',
];
