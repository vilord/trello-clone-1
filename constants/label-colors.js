module.exports = {
  GREEN: '#64BB55',
  YELLOW: '#F5DF63',
  ORANGE: '#FDAA54',
  RED: '#E95B4B',
  PURPLE: '#C27ADE',
  BLUE: '#127ABD',
  LIGHT_BLUE: '#20C2DE',
  LIGHT_GREEN: '#59E79A',
  PINK: '#FD83CD',
  DARK_GREY: '#4D4D4D',
};
