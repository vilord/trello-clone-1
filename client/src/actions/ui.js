import * as types from '../constants/actionTypes';

/**
 * UI Errors
 */
export const setUiError = error => ({
  type: types.SET_UI_ERROR,
  error,
});

export const resetUiError = () => ({
  type: types.RESET_UI_ERROR,
});

/**
 * Boards Explorer
 */
export const showBoardsExplorer = () => ({
  type: types.SHOW_BOARDS_EXPLORER,
});

export const hideBoardsExplorer = () => ({
  type: types.HIDE_BOARDS_EXPLORER,
});

/**
 * Header Search
 */
export const focusHeaderSearch = () => ({
  type: types.FOCUS_HEADER_SEARCH,
});

export const blurHeaderSearch = () => ({
  type: types.BLUR_HEADER_SEARCH,
});

export const setHeaderSearch = term => ({
  type: types.SET_HEADER_SEARCH,
  term,
});

/**
 * Create Menu
 */
export const showCreateMenu = () => ({
  type: types.SHOW_CREATE_MENU,
});

export const hideCreateMenu = () => ({
  type: types.HIDE_CREATE_MENU,
});

/**
 * Notifications
 */
export const showNotifications = () => ({
  type: types.SHOW_NOTIFICATIONS,
});

export const hideNotifications = () => ({
  type: types.HIDE_NOTIFICATIONS,
});

/**
 * User Menu
 */
export const showUserMenu = () => ({
  type: types.SHOW_USER_MENU,
});

export const hideUserMenu = () => ({
  type: types.HIDE_USER_MENU,
});

/**
 * Rename Board
 */
export const showRenameBoard = () => ({
  type: types.SHOW_RENAME_BOARD,
});

export const hideRenameBoard = () => ({
  type: types.HIDE_RENAME_BOARD,
});

/**
 * Add to Team
 */
export const showAddToTeam = () => ({
  type: types.SHOW_ADD_TO_TEAM,
});

export const hideAddToTeam = () => ({
  type: types.HIDE_ADD_TO_TEAM,
});

/**
 * Visibility Menu
 */
export const showVisibilityMenu = () => ({
  type: types.SHOW_VISIBILITY_MENU,
});

export const hideVisibilityMenu = () => ({
  type: types.HIDE_VISIBILITY_MENU,
});

/**
 * Board Menu
 */
export const showBoardMenu = () => ({
  type: types.SHOW_BOARD_MENU,
});

export const hideBoardMenu = () => ({
  type: types.HIDE_BOARD_MENU,
});
