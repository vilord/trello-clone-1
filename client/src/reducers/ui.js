import * as types from '../constants/actionTypes';
import initState from './initUiState';

const uiReducer = (state = initState, action) => {
  switch (action.type) {
    case types.SET_UI_ERROR:
      return {
        ...state,
        error: action.error,
      };
    case types.RESET_UI_ERROR:
      return {
        ...state,
        error: {
          kind: '',
          header: '',
          message: '',
        },
      };
    case types.FOCUS_HEADER_SEARCH:
      return {
        ...state,
        focus: 'search',
      };
    case types.BLUR_HEADER_SEARCH:
      return {
        ...state,
        focus: '',
      };
    case types.SIGNUP_USER_REQUEST:
      return {
        ...state,
        fetching: {
          ...state.fetching,
          signup: true,
        },
      };
    case types.SIGNUP_USER_SUCCESS:
      return {
        ...state,
        fetching: {
          ...state.fetching,
          signup: false,
        },
      };
    case types.SIGNUP_USER_FAILURE:
      return {
        ...state,
        error: action.error,
        fetching: {
          ...state.fetching,
          signup: false,
        },
      };
    case types.LOGIN_USER_REQUEST:
    case types.GET_USER_SESSION_REQUEST:
      return {
        ...state,
        fetching: {
          ...state.fetching,
          login: true,
        },
      };
    case types.LOGIN_USER_SUCCESS:
    case types.GET_USER_SESSION_SUCCESS:
      return {
        ...state,
        fetching: {
          ...state.fetching,
          login: false,
        },
      };
    case types.LOGIN_USER_FAILURE:
      return {
        ...state,
        error: action.error,
        fetching: {
          ...state.fetching,
          login: false,
        },
      };
    case types.LOGOUT_USER:
      return {
        ...state,
        fetching: {
          ...state.fetching,
          login: false,
        },
      };
    default:
      return state;
  }
};

export default uiReducer;
