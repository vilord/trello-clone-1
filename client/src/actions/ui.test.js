import * as actions from './ui';
import * as types from '../constants/actionTypes';
import * as errors from '../constants/errors';

describe('ui actions', () => {
  beforeAll(() => {
    console.error = err => {
      throw new Error(err);
    };
  });

  describe('Action Creators', () => {
    describe('UI Errors', () => {
      it('creates a SET_UI_ERROR action', () => {
        const expected = {
          type: types.SET_UI_ERROR,
          error: errors.invalidEmail,
        };

        expect(actions.setUiError(errors.invalidEmail)).toEqual(expected);
        expect(actions.setUiError(errors.invalidEmail)).toMatchSnapshot();
      });

      it('creates a RESET_UI_ERROR action', () => {
        const expected = { type: types.RESET_UI_ERROR };
        expect(actions.resetUiError()).toEqual(expected);
        expect(actions.resetUiError()).toMatchSnapshot();
      });
    });

    describe('Main Menu', () => {
      it('creates a SHOW_MAIN_MENU action', () => {
        const expected = { type: types.SHOW_MAIN_MENU };
        expect(actions.showMainMenu()).toEqual(expected);
        expect(actions.showMainMenu()).toMatchSnapshot();
      });

      it('creates a HIDE_MAIN_MENU action', () => {
        const expected = { type: types.HIDE_MAIN_MENU };
        expect(actions.hideMainMenu()).toEqual(expected);
        expect(actions.hideMainMenu()).toMatchSnapshot();
      });
    });

    describe('Boards Menu', () => {
      it('creates a SHOW_BOARDS_MENU action', () => {
        const expected = { type: types.SHOW_BOARDS_MENU };
        expect(actions.showBoardsMenu()).toEqual(expected);
        expect(actions.showBoardsMenu()).toMatchSnapshot();
      });

      it('creates a HIDE_BOARDS_MENU action', () => {
        const expected = { type: types.HIDE_BOARDS_MENU };
        expect(actions.hideBoardsMenu()).toEqual(expected);
        expect(actions.hideBoardsMenu()).toMatchSnapshot();
      });
    });

    describe('Header Search', () => {
      it('creates a FOCUS_HEADER_SEARCH action', () => {
        const expected = { type: types.FOCUS_HEADER_SEARCH };
        expect(actions.focusHeaderSearch()).toEqual(expected);
        expect(actions.focusHeaderSearch()).toMatchSnapshot();
      });

      it('creates a BLUR_HEADER_SEARCH action', () => {
        const expected = { type: types.BLUR_HEADER_SEARCH };
        expect(actions.blurHeaderSearch()).toEqual(expected);
        expect(actions.blurHeaderSearch()).toMatchSnapshot();
      });
    });
  });
});
