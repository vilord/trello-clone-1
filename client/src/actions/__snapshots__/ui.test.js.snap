// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui actions Action Creators Boards Menu creates a HIDE_BOARDS_MENU action 1`] = `
Object {
  "type": "HIDE_BOARDS_MENU",
}
`;

exports[`ui actions Action Creators Boards Menu creates a SHOW_BOARDS_MENU action 1`] = `
Object {
  "type": "SHOW_BOARDS_MENU",
}
`;

exports[`ui actions Action Creators Header Search creates a BLUR_HEADER_SEARCH action 1`] = `
Object {
  "type": "BLUR_HEADER_SEARCH",
}
`;

exports[`ui actions Action Creators Header Search creates a FOCUS_HEADER_SEARCH action 1`] = `
Object {
  "type": "FOCUS_HEADER_SEARCH",
}
`;

exports[`ui actions Action Creators Main Menu creates a HIDE_MAIN_MENU action 1`] = `
Object {
  "type": "HIDE_MAIN_MENU",
}
`;

exports[`ui actions Action Creators Main Menu creates a SHOW_MAIN_MENU action 1`] = `
Object {
  "type": "SHOW_MAIN_MENU",
}
`;

exports[`ui actions Action Creators UI Errors creates a RESET_UI_ERROR action 1`] = `
Object {
  "type": "RESET_UI_ERROR",
}
`;

exports[`ui actions Action Creators UI Errors creates a SET_UI_ERROR action 1`] = `
Object {
  "error": Object {
    "header": "Verify email!",
    "kind": "invalidEmail",
    "message": "That doesn't look like an email addressâ€¦",
  },
  "type": "SET_UI_ERROR",
}
`;
